File: BlogApp-RN/src/features/authentication/domain/repos/authRepo.ts
import { AsyncResult } from "../../../../core/typedefs/asyncResult";
import { AuthResponse } from "../../data/models/authResponse";
import { ResponseMessage } from "../../data/models/responseMessage";

abstract class AuthRepo {
  abstract login: (username: string, password: string) => AsyncResult<AuthResponse>;
  abstract logout: () => AsyncResult<ResponseMessage>;
  abstract signUp: (username: string, password: string) => AsyncResult<AuthResponse>;
}

export default AuthRepo;

File: blogapp_flutter/lib/features/authentication/domain/repos/auth_repo.dart
import '../../../../core/typedefs/async_result.dart';
import '../../data/models/auth_response.dart';
import '../../data/models/response_message.dart';

abstract class AuthRepo {
  AsyncResult<AuthResponse> login({
    required String username,
    required String password,
  });

  AsyncResult<ResponseMessage> logout();

  AsyncResult<AuthResponse> signUp({
    required String username,
    required String password,
  });
}